aws/src/io/pedestal/http/aws/lambda/utils.clj

```diff
@@ -0,0 +1,291 @@
+;;     It will perform the JSON parse automatically ---
+(defn direct-apigw-provider
+  "Given a service map, return a service map with a provider function
+  for an AWS API Gateway event, under `:io.pedestal.aws.lambda/apigw-handler`.
+
+  This provider function takes the apigw-event map and the runtime.Context
+  and returns an AWS API Gateway response map (containing -- :statusCode :body :headers)
+  You may want to add a custom interceptor in your chain to handle Scheduled Events.
+
+  This chain terminates if a Ring `:response` is found in the context
+  or an API Gateway `:apigw-response` map is found.
+
+  All additional conversion, coercion, writing, and extension should be handled by
+  interceptors in the interceptor chain."
+  [service-map]
+  (let [interceptors (:io.pedestal.http/interceptors service-map [])
+        default-context (get-in service-map [:io.pedestal.http/container-options :default-context] {})
+        body-processor (get-in service-map
+                               [:io.pedestal.http/container-options :body-processor]
+                               (resolve-body-processor))]
+    (assoc service-map
+           :io.pedestal.aws.lambda/apigw-handler
+           (fn [apigw-event ^Context context] ;[^InputStream input-stream ^OutputStream output-stream ^Context context]
+             (let [;event (json/parse-stream
+                   ;        (java.io.PushbackReader. (java.io.InputStreamReader. input-stream))
+                   ;        nil
+                   ;        nil)
+                   request (apigw-request-map apigw-event)
+                   initial-context (merge {;:aws.lambda/input-stream input-stream
+                                           ;:aws.lambda/output-stream output-stream
+                                           :aws.lambda/context context
+                                           :aws.apigw/event apigw-event
+                                           :request request
+                                           ::chain/terminators [#(let [resp (:response %)]
+                                                                   (and (map? resp)
+                                                                        (integer? (:status resp))
+                                                                        (map? (:headers resp))))
+                                                                #(map? (:apigw-response %))]}
+                                          default-context)
+                   response-context (chain/execute initial-context interceptors)
+                   response-map (or (:apigw-response response-context)
+                                    ;; Use `or` to prevent evaluation
+                                    (some-> (:response response-context)
+                                            (apigw-response body-processor)))]
+               response-map)))))
+
+
+;(defprotocol IntoRequestStreamHandler
```

Comment:
Currently, the public API is only what has been used in production settings.  What is commented out provides the foundation for supporting Pedestal at the Servlet level within Lambda (and some general utilities that go along with that).  I elected to leave the code commented out rather than delete it, so we can eventually work toward that goal (or so that individuals can roll their own support, if they need)